### PR TITLE
add gofmt to test github action

### PR DESCRIPTION
**Pull request by J-Thompson12**
> Describe what has changed in this PR 

Closes at #527 Added gofmt check to github action test section

> Tell your future self why have you made these changes

To make sure all code is formatted

> How have you verified this change? Tested locally? Added unit test/integration/acceptance test(s)? 

using github actions

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -43,6 +43,8 @@ jobs:
       with:
         version: v1.29
         args: --timeout=600s
+    - name: Format
+      run: if [ "$(gofmt -l . | wc -l)" -gt 0 ]; then exit 1; fi
```
Comment (jpellizzari) [inline .github/workflows/test.yml:47]: What does the output look like if this fails?

Comment (J-Thompson12) [reply .github/workflows/test.yml:47]: If the condition hits it just fails as exit code 1 but its the only thing run there. I realize you are right i should output the files that need formatting.

**Comment by jpellizzari** [comment]
@J-Thompson12 One more thing I forgot to add: can you encapsulate this in a `make` target? It would be reproduce locally.

**Review by jpellizzari — APPROVED** [review]
Nice work!

**Pull request merged**